### PR TITLE
more cleanup

### PR DESCRIPTION
**Pull request by BenTheElder**
- improve some code comments / add more
- trim the metadata we set for kustomize patching from the generated kubeadm config output before using it

### CODE REVIEW

**Comment by BenTheElder** [comment]
ah, older Go does not have strings.ReplaceAll, switched to strings.Replace
with -1

File: pkg/cluster/internal/create/actions/config/config.go [hunk 1]
```diff
@@ -111,7 +112,24 @@ func getKubeadmConfig(cfg *config.Cluster, data kubeadm.ConfigData) (path string
...
+	patched, err := kustomize.Build([]string{config}, patches, jsonPatches)
+	if err != nil {
+		return "", err
+	}
+	return removeMetadata(patched), nil
+}
+
+// trims out the metadata.name we put in the config for kustomize matching,
+// kubeadm will complain about this otherwise
+func removeMetadata(kustomized string) string {
```
Comment (neolit123) [inline pkg/cluster/internal/create/actions/config/config.go]: nice, this should get rid of the unmarshal warnings.

File: pkg/container/docker/archive.go [hunk 2]
```diff
@@ -77,8 +77,17 @@ func GetArchiveTags(path string) ([]string, error) {
...
 }
 
-// EditArchiveRepositories applies edit to reader's repositories file
-func EditArchiveRepositories(reader io.Reader, writer io.Writer, edit func(string) string) error {
+// EditArchiveRepositories applies edit to reader's image repositories,
+// IE the repository part of repository:tag in image tags
+// This supports v1 / v1.1 / v1.2 Docker Image Archives
+//
+// editRepositories should be a function that returns the input or an edited
+// form, where hte input is the image repository
```
Comment (neolit123) [inline pkg/container/docker/archive.go]: typo hte

Comment (BenTheElder) [reply pkg/container/docker/archive.go]: fixed

**Pull request merged**